### PR TITLE
Remove FlatSpatialPooler

### PR DESCRIPTION
**Pull request by david-ragazzi**
Fixes: https://github.com/numenta/nupic/issues/627

### CODE REVIEW

File: tests/unit/nupic/research/spatial_pooler_unit_test.py [hunk 1]
```diff
@@ -1789,6 +1789,56 @@ def testWrite(self):
     self.assertSetEqual(indices1, indices2)
 
 
+  def testRandomSPDoesNotLearn(self):
```
Comment (david-ragazzi) [inline tests/unit/nupic/research/spatial_pooler_unit_test.py]: This test was ported from FlatSP and works fine.

**Comment by david-ragazzi** [comment]
CC @chetan51

**Comment by david-ragazzi** [comment]
@chetan51 Ping...

**Comment by chetan51** [comment]
:+1:

File: tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py [hunk 2]
```diff
@@ -1,533 +0,0 @@
...
-        val,
-        blist[idx],
-        places = 3,
-        msg = "Lists different at index %d with values %g and %g"
-              % (idx, val, blist[idx]))
-
-
-  def compare(self, pySp, cppSp):
-    self.assertAlmostEqual(pySp.getNumColumns(),
-                           cppSp.getNumColumns())
```
Comment (breznak) [inline tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py]: I think this c++/py compatibility test should stay! (ofc ported to the new SP implementation, not FlatSP)

File: tests/unit/nupic/research/flat_spatial_pooler_cpp_api_test.py [hunk 3]
```diff
@@ -1,34 +0,0 @@
...
-
-from nupic.bindings.algorithms import FlatSpatialPooler as CPPFlatSpatialPooler
-
-import flat_spatial_pooler_py_api_test
-
-flat_spatial_pooler_py_api_test.SpatialPooler = CPPFlatSpatialPooler
-FlatSpatialPoolerCPPAPITest = flat_spatial_pooler_py_api_test.FlatSpatialPoolerAPITest
-
-if __name__ == "__main__":
-  unittest.main()
```
Comment (breznak) [inline tests/unit/nupic/research/flat_spatial_pooler_cpp_api_test.py]: same, c++ API test should be ported to new pySP, but not removed

**Comment by breznak** [comment]
I'm raising a finger of caution on the removed tests, thoughts @david-ragazzi @chetan51 ?
```
 delete mode 100644 nupic/research/flat_spatial_pooler.py
 delete mode 100755 tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py
 delete mode 100644 tests/unit/nupic/research/flat_spatial_pooler_cpp_api_test.py
 delete mode 100644 tests/unit/nupic/research/flat_spatial_pooler_py_api_test.py
 delete mode 100755 tests/unit/nupic/research/flat_spatial_pooler_unit_test.py

```

File: tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py (hunk 2 above)
Comment (david-ragazzi) [inline tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py]: > I think this c++/py compatibility test should stay! (ofc ported to the new SP implementation, not FlatSP)

Relax, man.. :relieved:  Look these here:

https://github.com/numenta/nupic/blob/master/tests/unit/nupic/research/spatial_pooler_compatability_test.py#L58

Comment (breznak) [inline tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py]: cool then! :sunglasses: 
Sorry, I just saw the files removed in the recent pull, and none tests added...

**Pull request merged**